uvtools/dspec.py

```diff
@@ -109,7 +109,7 @@ def calc_width(filter_size, real_delta, nsamples):
     return (uthresh, lthresh)
 
 def fourier_filter(x, data, wgts, filter_centers, filter_half_widths, suppression_factors,
```

Comment:
I think the `dayenu` and `fit_basis` modes of `fourier_filter` are fairly streamlined. The real trouble is in the `clean` functionality. ideally, this function would call `vis_clean` to do the cleaning, but unfortunately, `vis_clean` does not support filtering areas that are not centered at delay=0. I think the right way forward would be to generalize `vis_clean` so that it can handle off-center cleaning windows and then have `fourier_filter` call it. I will make an issue so that this can be achieved in a future PR.